### ISSUE TITLE
Allow URI for mongoDB connection

### ISSUE THREAD

**Issue by saitensou** [tinode/chat#755]
**Is your feature request related to a problem? Please describe.**
I was trying to use Tinode with mongoDB atlas, which only allows connections using connection string URL.
From the current config I don't see a clear/easy way to incorporate with it

**Describe the solution you'd like**
Allow the URI option in tinode.conf

**Describe alternatives you've considered**
Not much, the connection given at the atlas gives the connection string as
> mongodb+srv://__DB_NAME__.mongodb.net/__PATH_AND_PARAMS__
There is not much to do in current configs allowed

**Comment by or-else** [comment]
Can you explain the need for this:

```
serverAPIOptions := mdbopts.ServerAPI(mdbopts.ServerAPIVersion1)
opts.SetServerAPIOptions(serverAPIOptions)
```

Why is it needed? Why is it needed only when the connection string URI is used?

**Comment by saitensou** [comment]
I am not very sure on the purpose of this too, as it is specified in nowhere in the documents except the connection guide steps.
From the official document it `represent options used to configure the API version sent to the server`, which I think is only needed when using atlas. 
Plus some trial and error on the connection, this is only needed when using Atlas (connection string URI).

Not sure if binding it with the whole `uri` option is the best, but I think it is a sufficient one as "usual mongodb connection" can be tackled using the original, detailed configs instead.

**Comment by tairov** [comment]
Regarding Mongodb Atlas, there could be compatibility issues related to the DB version. Need to carefully test out all the original queries from chat-server.
Here is DNS Seed List Connection Format documentation : https://www.mongodb.com/docs/manual/reference/connection-string/#dns-seed-list-connection-format

**Comment by or-else** [comment]
Would you do the testing?

**Comment by saitensou** [comment]
@tairov Are you stating that there could be compatibility issue with the DNS Seed List Connection Format?

My project does not use Seed List Connection Format but the URL provided from Mongo Guide is somewhat alike
> mongodb+srv://_CREDENTIALS_@_PROJECT_.gmuaq.mongodb.net/_DATABASE_?retryWrites=true&w=majority

Thus I think it also supports using DNS Seed List Connection Format.
If there are compatibility Issue, I am not sure if it is related to ServerAPIVersion, or other settings. 
But as long as the work around can be set using `ApplyURI` api it would be fine with the fix I suppose. 
@or-else what do you think?

[Note: Currently official docs supports only ServerAPIVersion 1](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo/options?utm_source=godoc#ServerAPIOptions)

**Comment by or-else** [comment]
Please send a pull request if you need any changes in the driver.

**Comment by or-else** [comment]
The 'Allow URI for mongoDB connection' request is released in 0.19.0. I'm going to close this issue. Please feel free to open another issue if you feel more work is needed.
